pkgs/build-support/trivial-builders/default.nix

```diff
@@ -591,6 +591,43 @@ rec {
     ln -s ${lib.getBin drv}/bin $out/bin
   '';
 
+  /*
+    Adds a hook to the nativeBuildInputs of a derivation which will provide an
+    SSH session via tmate to the build environment, for interactive
+    debugging
+
+    Example:
+
+    # Spawn a tmate session after the `buildPhase` of the `hello` derivation
+    addTmateBreakpoint (pkgs.hello.overrideAttrs { buildPhase = "exit 1;"; })
+  */
```

Comment:
Could you move this into the manual?
See e.g. https://nixos.org/manual/nixpkgs/stable/index.html#breakpointhook, which would be good to link back and forth.